packages/client/src/helpers.ts

```diff
@@ -123,6 +126,25 @@ export function rethrowKnownErrorTypes(err: GrpcServiceError): void {
         const { namespace } = temporal.api.errordetails.v1.NamespaceNotFoundFailure.decode(entry.value);
         throw new NamespaceNotFoundError(namespace);
       }
+      case 'temporal.api.errordetails.v1.MultiOperationExecutionFailure': {
+        // MultiOperationExecutionFailure contains error statuses for multiple
+        // operations. A MultiOperationExecutionAborted error status means that
+        // the corresponding operation was aborted due to an error in one of the
+        // other operations. We rethrow the first operation error that is not
+        // MultiOperationExecutionAborted.
+        const { statuses } = temporal.api.errordetails.v1.MultiOperationExecutionFailure.decode(entry.value);
+        for (const status of statuses) {
+          const detail = status.details?.[0];
+          const statusType = detail?.type_url?.replace(/^type.googleapis.com\//, '') as FailureName | undefined;
+          if (statusType === 'temporal.api.failure.v1.MultiOperationExecutionAborted' || status.code === Status.OK) {
+            continue;
+          }
+          err.message = status.message ?? err.message;
```

Comment:
Need to check for some specific types of inner errors and throw the appropriate classes in those cases.

Reply:
I believe we agreed that these are all done.